kubespawner/reflector.py

```diff
@@ -27,6 +28,15 @@ class ResourceReflector(LoggingConfigurable):
     kubernetes resources.
 
     Must be subclassed once per kind of resource that needs watching.
+
+    Creating a reflector should be done with the create() classmethod,
+    since that, in addition to creating the instance, initializes the
+    Kubernetes configuration, acquires a K8s API client, and starts the
+    watch task.
+
+    Shutting down a reflector should be done by awaiting its stop() method.
+    JupyterHub does not currently do this, so the watch task runs until the
+    Hub exits.
```

Comment:
I figure JupyterHub shouldn't do this, its not aware of k8s or reflectors etc, only KubeSpawner is, so its a KubeSpawner topic only. The question is, should we make KubeSpawner do this? Maybe. If so, how? Via an async destructor/finalizer?

```suggestion
    Shutting down a shared reflector should be done by awaiting its stop()
    method. KubeSpawner doesn't gracefully do this currently, so the watch
    task runs until the Hub exits.
```